### PR TITLE
Added oscap_policy related test-cases

### PR DESCRIPTION
**Pull request by kbidarkar**
a) This also includes few doc-strings fix.

### CODE REVIEW

File: tests/foreman/cli/test_activationkey.py [hunk 1]
```diff
@@ -765,7 +765,7 @@ def test_positive_usage_limit(self):
         2. Update Usage Limit to a finite number
         3. Register Content hosts to match the Usage Limit
         4. Attempt to register an other Content host after reaching the Usage
-        Limit
+           Limit
```
Comment (kbidarkar) [inline tests/foreman/cli/test_activationkey.py]: Fix required for `make docs-clean docs` to PASS.

File: robottelo/ui/oscappolicy.py [hunk 2]
```diff
@@ -27,6 +27,8 @@ def create(self, name, desc=None, content=None, profile=None, period=None,
         if desc:
             self.text_field_update(locators['oscap.desc_policy'], desc)
         self.click(common_locators['submit'])
+        if self.wait_until_element(common_locators['haserror']):
+            return
```
Comment (kbidarkar) [inline robottelo/ui/oscappolicy.py]: This is added to handle `neagtive_create` scenario.  This page cannot be submitted in ONE GO and hence this check.

**Comment by kbidarkar** [comment]
Following are the test results:
```
(robottelo)[kbidarka@dhcp201-208 robottelo]$ nosetests -s -v tests.foreman.ui.test_oscappolicy
....F

----------------------------------------------------------------------
Ran 4 tests in 1620.268s

FAILED (errors=1)
```
Re-running the failed test after fixing it.
```
(robottelo)[xyz@abc robottelo]$ nosetests -s -v tests.foreman.ui.test_oscappolicy

ok

----------------------------------------------------------------------
Ran 1 test in 144.037s

OK
```

File: docs/api/tests.foreman.cli.rst [hunk 3]
```diff
@@ -223,11 +223,6 @@
 
 .. automodule:: tests.foreman.cli.test_syncplan
 
-:mod:`tests.foreman.cli.test_system_registration`
--------------------------------------------------
-
-.. automodule:: tests.foreman.cli.test_system_registration
-
```
Comment (kbidarkar) [inline docs/api/tests.foreman.cli.rst]: Being removed as now `tests.foreman.cli.test_system_registration` is renamed as `tests.foreman.cli.test_contenthost`.

**Comment by kbidarkar** [comment]
Here is the docs test.
```
(robottelo)[xyz@abc robottelo]$ make docs-clean docs
make[1]: Entering directory '~/git_world/robottelo/docs'
rm -rf _build/*
make[1]: Leaving directory '~/git_world/robottelo/docs'
make[1]: Entering directory '~/git_world/robottelo/docs'
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v1.3.1
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 19 source files that are out of date
updating environment: 19 added, 0 changed, 0 removed
reading sources... [100%] reviewing_PRs                                                                                                                                                                            
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] reviewing_PRs                                                                                                                                                                             
generating indices... genindex py-modindex
highlighting module code... [100%] robottelo.datafactory                                                                                                                                                           
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded.

Build finished. The HTML pages are in _build/html.
make[1]: Leaving directory '~/git_world/robottelo/docs'
```

**Comment by kbidarkar** [comment]
This PR is ready for review.

File: robottelo/ui/oscappolicy.py (hunk 2 above)
Comment (elyezer) [inline robottelo/ui/oscappolicy.py]: AFAIK other UI methods raise an exception if a problem is found. Would be better to use the same approach here?

Comment (elyezer) [inline robottelo/ui/oscappolicy.py]: Like https://github.com/SatelliteQE/robottelo/pull/3177/files#diff-7a561727bcf0c742d57c615b4d9a7bc0R63

**Comment by elyezer** [comment]
It looks good to my UI skills. ACK, pending comment.

File: docs/api/tests.foreman.cli.rst (hunk 3 above)
Comment (sthirugn) [inline docs/api/tests.foreman.cli.rst]: :+1:

File: tests/foreman/ui/test_oscappolicy.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+    @classmethod
+    def setUpClass(cls):
+        super(OpenScapPolicy, cls).setUpClass()
+        cls.content_path = get_data_file(
+            settings.oscap.content_path
+        )
+
+    @tier1
+    def test_positive_create(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_oscappolicy.py]: test_positive_create_with_policy_name?

File: tests/foreman/ui/test_oscappolicy.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+                        name=policy_name,
+                        period=OSCAP_PERIOD['weekly'],
+                        profile=OSCAP_PROFILE['rhccp'],
+                        period_value=OSCAP_WEEKDAY['friday'],
+                    )
+                    self.assertIsNotNone(
+                        self.oscappolicy.search(policy_name))
+
+    @tier1
+    def test_positive_delete(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_oscappolicy.py]: test_positive_delete_with_policy_name?

File: tests/foreman/ui/test_oscappolicy.py [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+                        profile=OSCAP_PROFILE['rhccp'],
+                        period_value=OSCAP_WEEKDAY['friday'],
+                    )
+                    self.assertIsNotNone(
+                        self.oscappolicy.search(policy_name))
+                    self.oscappolicy.delete(policy_name)
+
+    @skip_if_bug_open('bugzilla', 1293296)
+    @tier1
+    def test_negative_create_with_invalid_name(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_oscappolicy.py]: test_negative_create_with_policy_name?

File: robottelo/ui/oscappolicy.py (hunk 2 above)
Comment (kbidarkar) [inline robottelo/ui/oscappolicy.py]: This is unique case, where in we move from one form to another ( verification of the form happens one by one) and only in the end that it submits. So there is no way I can think of currently which can be done, as we used to do earlier or for others.

What this does is, it returns if the `general` info form has any issues and asserts that the entity is indeed None ( not created due to the negative scenarios) .

File: tests/foreman/ui/test_oscappolicy.py (hunk 6 above)
Comment (kbidarkar) [inline tests/foreman/ui/test_oscappolicy.py]: will leave this as it is for consistency sake, will update the others. :)

**Comment by sthirugn** [comment]
ACK

**Pull request merged**